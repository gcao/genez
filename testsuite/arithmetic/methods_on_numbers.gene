#!/usr/bin/env gene run

# Test: Methods on numbers
(var a 10)
(var b 20)

(print "a .+ b = " (a .+ b))
(print "b .- a = " (b .- a))
(print "a .* b = " (a .* b))
(print "b ./ a = " (b ./ a))

# Comparison methods
(print "a .< b = " (a .< b))
(print "b .> a = " (b .> a))
(print "a .== a = " (a .== a))
(print "a .== b = " (a .== b))