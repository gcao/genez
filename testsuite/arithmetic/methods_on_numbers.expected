a .+ b =  30
b .- a =  10
a .* b =  200
b ./ a =  2
a .< b =  true
b .> a =  true
a .== a =  true
a .== b =  false